plugins/cpp/parser/src/symbolhelper.cpp

```diff
@@ -64,127 +54,20 @@ const clang::Type* getStrippedType(const clang::QualType& qt_)
-      ? "anonymous-ns:" + std::to_string(fileLoc_.file.object_id())
-      : nd_->getQualifiedNameAsString();
-  }
-  else if (const clang::CXXRecordDecl* rd
-    = llvm::dyn_cast<clang::CXXRecordDecl>(nd_))
-  {
-    // AST nodes of type CXXRecordDecl contain another CXXRecordDecl node. The
-    // outer one is for the definition (and its position is the whole
-    // definition) and the inner one is for a declaration (of which the position
-    // is the "class" keyword and the class name.
-    // The qualified name of the inner node is ClassName::ClassName, although we
-    // should choose the same mangled name i.e. the qualified name of its
-    // parent.
-
-    const clang::CXXRecordDecl* parent
-      = llvm::dyn_cast<clang::CXXRecordDecl>(rd->getParent());
-
-    return parent && parent->getName() == rd->getName()
-      ? parent->getQualifiedNameAsString()
-      : rd->getQualifiedNameAsString();
-  }
-
-  // TODO: For some reason some named decls don't have name.
-  // Function parameters without name (e.g. compiler generated functions) don't
-  // have a name, but we can generate them a unique name later.
-  if (nd_->getNameAsString().empty() && !llvm::isa<clang::FunctionDecl>(nd_))
-    return nd_->getQualifiedNameAsString();
-
-  // No need to mangle C sources or declarations in extern "C".
-  if (!mangleContext_->shouldMangleDeclName(nd_))
-    return nd_->getQualifiedNameAsString();
-
-  const clang::FunctionDecl* fd = llvm::dyn_cast<clang::FunctionDecl>(nd_);
-  if (fd && llvm::isa<clang::FunctionProtoType>(
-    fd->getType()->getAs<clang::FunctionType>()))
-  {
-    // FIXME: Workaround for a crash reported in #236
-    // From Clang's code:
-    // We should never mangle anything without a prototype.
-
-    std::string str;
-    llvm::raw_string_ostream out(str);
-
-    if (const clang::CXXConstructorDecl* ctor
-        = llvm::dyn_cast<clang::CXXConstructorDecl>(fd))
-      mangleContext_->mangleCXXCtor(ctor, clang::Ctor_Complete, out);
-    else if (const clang::CXXDestructorDecl* dtor
-        = llvm::dyn_cast<clang::CXXDestructorDecl>(fd))
-      mangleContext_->mangleCXXDtor(dtor, clang::Dtor_Complete, out);
-    else
-      mangleContext_->mangleName(fd, out);
-    
-    // TODO: In the case when a function has a template and a non-template
-    // version then their mangled name is the same as long as the template is
-    // not instantiated. Distinguising them by their locations is not good,
-    // because if there is a function which has two versions in an #ifdef ...
-    // #else ... #endif section then every usage will also have two versions.
-    // So if a function has a template and a non-template version at the same
-    // time then querying the definition of the non-template usage the template
-    // version also returns.
-    return out.str();
-  }
-
-  return nd_->getQualifiedNameAsString();
+  llvm::SmallVector<char, 64> usr;
+  clang::index::generateUSRForDecl(nd_, usr);
```

Comment:
`generateUSRFor...` functions return a bool which are signalling whether "the result of the call should be ignored".

Reply:
Thanks, I haven't noticed the return type. I'll check what it means and in what cases it returns true or false.

Reply:
We investigated a little the return value of this function. I put a relatively long comment in the source code which describes why we don't use this value:
https://github.com/Ericsson/CodeCompass/pull/429/files#diff-b735f57c014d7a4293c17acf122b284dR60-R82
I don't know if this explanation is satisfactory.